packages/react-instantsearch-core/src/core/__tests__/createInstantSearchManager.js

```diff
@@ -150,6 +150,237 @@ describe('createInstantSearchManager', () => {
     });
   });
 
+  describe('getSearchParameters', () => {
+    it('expect a widget top level to be shared between main and derived parameters', () => {
+      // <InstantSearch indexName="index">
+      //   <SearchBox defaultRefinement="shared" />
+      //   <Index indexId="main" indexName="main" />
+      // </InstantSearch>
+
+      const ism = createInstantSearchManager({
+        indexName: 'index',
+        searchClient: createSearchClient(),
+      });
+
+      // <SearchBox defaultRefinement="shared" />
+      ism.widgetsManager.registerWidget({
+        getSearchParameters(state) {
+          return state.setQuery('shared');
+        },
+        context: {},
+        props: {},
+      });
+
+      // <Index indexId="main" indexName="main" />
+      ism.widgetsManager.registerWidget({
+        getSearchParameters(state) {
+          return state.setIndex('main');
+        },
+        props: {
+          indexId: 'main',
+        },
+      });
+
+      const { mainParameters, derivedParameters } = ism.getSearchParameters();
+
+      expect(mainParameters).toEqual(
+        expect.objectContaining({
+          index: 'index',
+          query: 'shared',
+        })
+      );
+
+      expect(derivedParameters).toEqual([
+        {
+          indexId: 'main',
+          parameters: expect.objectContaining({
+            index: 'main',
+            query: 'shared',
+          }),
+        },
+      ]);
+    });
+
+    it('expect a widget with the same id than the indexName to be a main parameters', () => {
```

Comment:
is it not a little confusing that `indexId` on the `<Index />` component maps to `indexName` on `<InstantSearch />`, while  `<Index />` has an `indexName` prop, too?

Reply:
Yes it's intended. We test that the actual value of the index that we want to target is the `indexName` of the `Index` component not the one of the `InstantSearch` component.